core/src/kura.rs

```diff
@@ -166,13 +174,23 @@ impl Kura {
 
         info!("Loaded {} blocks at init.", blocks.len());
 
-        let hash_array = blocks.iter().map(VersionedCommittedBlock::hash).collect();
+        {
+            let hash_array = blocks.iter().map(VersionedCommittedBlock::hash).collect();
+
+            let mut guard = self
+                .block_hash_array
+                .lock()
+                .expect("lock on block hash array");
+            *guard = hash_array;
+        }
+        {
+            let mut guard = self
+                .block_arc_array
+                .lock()
+                .expect("lock on block arc array");
 
-        let mut guard = self
-            .block_hash_array
-            .lock()
-            .expect("lock on block hash array");
-        *guard = hash_array;
+            *guard = vec![None; blocks.len()];
```

Comment:
This value is kind of thing that should be documented with comments. Or reconsidered…